modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java

```diff
@@ -150,4 +182,22 @@ protected void resolveLobs(String typeName, Object[] attributes) throws SQLExcep
         }
     }
 
+    public String getNestedTableType(String typeName) throws SQLException
+    {
+        String dataType = null;
+
+        try (PreparedStatement ps = this.prepareStatement(QUERY_ALL_COLL_TYPES))
+        {
+            ps.setString(1, typeName);
+
+            ResultSet resultSet = ps.executeQuery();
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Use try-with-resources or close this "ResultSet" in a "finally" clause. ([squid:S2095](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2095))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7681&issues=AWjt-QE0otIXnmsUG4vf&open=AWjt-QE0otIXnmsUG4vf)

[ISSUE_KEY]: AWjt-QE0otIXnmsUG4vf

Reply:
Take this into account.